src/lib/config.ts

```diff
@@ -1270,11 +1273,38 @@ export function unset(...target) {
 export async function save(storage: "local" | "sync" = get("storageloc")) {
     // let storageobj = storage === "local" ? browser.storage.local : browser.storage.sync
     // storageobj.set({CONFIGNAME: USERCONFIG})
+    await locks.acquire("config")
     const settingsobj = o({})
     settingsobj[CONFIGNAME] = USERCONFIG
-    return storage === "local"
+    const ans = storage === "local"
         ? browser.storage.local.set(settingsobj)
         : browser.storage.sync.set(settingsobj)
+
+    // Ignore errors about background pages that aren't listening
+    const friendly_bg_messenger = async () => {
+        let p
+        try {
+            p = await browser.runtime.sendMessage({type: "config", command: "confUpdate", args: [settingsobj]})
+        } catch (e) {
+            if (e.message != "Could not establish connection. Receiving end does not exist.") {
+                throw e
+            }
+            p = true
+        }
+        return p
+    }
+
+    await Promise.all([
+        // dispatch message to all content state.ts's
+        messaging.messageAllTabs("config", "confUpdate", [settingsobj]),
```

Comment:
On the one hand I am still traumatised by how long this PR took and how much it broke Tridactyl on various platforms.

On the other hand the `escapehatch` logic we have for finding live Tridactyl tabs could salvage it, maybe?